### PR TITLE
cmd/snap-confine: reduce verbosity of debug and error messages

### PR DESCRIPTION
**Pull request by zyga**
This branch contains various tweaks to debugging and error messages from namespace
management code. This also prepares the code for managing multiple mount namespaces
and less from managing multiple types of namespaces (just in the terms of wording used).

There are several formatting changes to the code to avoid wrapping but no semantic
changes.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: cmd/snap-confine/ns-support.c [hunk 1]
```diff
@@ -443,27 +423,26 @@ static int sc_inspect_and_maybe_discard_stale_ns(int mnt_fd,
 	}
 	// If the namespace is up-to-date then we are done.
 	if (value == SC_DISCARD_NO) {
-		debug("the mount namespace is up-to-date and can be reused");
+		debug("NOTE: preserved mount namespace can be reused");
```
[the remark this answers is not in the document]
Comment (zyga) [reply cmd/snap-confine/ns-support.c]: I added them in the several places where essential choices are done. I think it's fine to drop for now but I would love to bring them back in the final version.

**Comment by zyga** [comment]
Thanks for the review. I'll apply the suggestions to land this quickly. Some of the strings will change soon and one of the suggestions are actually applied in another branch (the fork+if move).

File: cmd/snap-confine/ns-support.c [hunk 2]
```diff
@@ -131,53 +130,41 @@ void sc_reassociate_with_pid1_mount_ns(void)
...
 		}
-		die("cannot perform readlinkat() on the mount namespace file "
-		    "descriptor of the init process");
+		die("cannot read mount namespace identifier of pid 1");
 	}
 	memset(self_buf, 0, sizeof self_buf);
 	if (readlinkat(self_mnt_fd, "", self_buf, sizeof self_buf) < 0) {
-		die("cannot perform readlinkat() on the mount namespace file "
-		    "descriptor of the current process");
+		die("cannot read mount namespace identifier of this process");
```
Comment (stolowski) [inline cmd/snap-confine/ns-support.c]: Nitpick, I think s/this/current/ sounds better (but other than that +1 for the shorter message).

File: cmd/snap-confine/ns-support.c [hunk 3]
```diff
@@ -131,53 +130,41 @@ void sc_reassociate_with_pid1_mount_ns(void)
...
 	}
 	if (memcmp(init_buf, self_buf, sizeof init_buf) != 0) {
-		debug("the current process does not share mount namespace with "
-		      "the init process, re-association required");
-		// NOTE: we cannot use O_NOFOLLOW here because that file will always be a
+		debug("moving to mount namespace of pid 1");
```
Comment (stolowski) [inline cmd/snap-confine/ns-support.c]: Nice, the old message was really longish..

File: cmd/snap-confine/ns-support.c [hunk 4]
```diff
@@ -131,53 +130,41 @@ void sc_reassociate_with_pid1_mount_ns(void)
...
+		// We cannot use O_NOFOLLOW here because that file will always be a
 		// symbolic link. We actually want to open it this way.
 		int init_mnt_fd_real SC_CLEANUP(sc_cleanup_close) = -1;
-		init_mnt_fd_real = open("/proc/1/ns/mnt", O_RDONLY | O_CLOEXEC);
+		init_mnt_fd_real = open(path_pid_1, O_RDONLY | O_CLOEXEC);
```
Comment (stolowski) [inline cmd/snap-confine/ns-support.c]: Nice, good idea having path_pid_1.

File: cmd/snap-confine/ns-support.c [hunk 5]
```diff
@@ -131,53 +130,41 @@ void sc_reassociate_with_pid1_mount_ns(void)
...
+	}
+	if (sc_is_mount_ns_dir_private()) {
+		return;
+	}
+	if (mount(sc_ns_dir, sc_ns_dir, NULL, MS_BIND | MS_REC, NULL) < 0) {
+		die("cannot self-bind mount %s", sc_ns_dir);
+	}
+	if (mount(NULL, sc_ns_dir, NULL, MS_PRIVATE, NULL) < 0) {
+		die("cannot change propagation type to MS_PRIVATE in %s",
+		    sc_ns_dir);
```
Comment (stolowski) [inline cmd/snap-confine/ns-support.c]: Nice, this block is now much more readable and compact.

File: cmd/snap-confine/ns-support.c [hunk 6]
```diff
@@ -376,22 +362,21 @@ static int sc_inspect_and_maybe_discard_stale_ns(int mnt_fd,
...
 				abort();
 			default:
-				die("cannot ensure that parent process is still alive");
+				die("cannot confirm that parent process is alive");
 				break;
 			}
 		}
 
-		debug("joining the namespace that we are about to probe");
+		debug("joining preserved mount namespace (for inspection)");
```
Comment (stolowski) [inline cmd/snap-confine/ns-support.c]: Nitpick, I'd drop quotes.

**Pull request merged**